### DISCUSSION TITLE
preview button in md/html/latex and svg files

### DISCUSSION THREAD

**Discussion by JustFrederik** [zed-industries/zed#26077] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

A button to toggle the preview. I know there is a preview for markdown over the command palette, but it would be great if it could be opened over a specific button. Monitor file changes and serve a directory isn't hard for html files. the svg images get displayed and only need a toggle. latex files can be rendered by [standalone](https://docs.rs/tectonic/latest/tectonic/#) or a [wrapper](https://docs.rs/texrender/latest/texrender/).

csv would be great too, but this would need a table renderer

### If applicable, add mockups / screenshots to help present your vision of the feature

Example from Jetbrains:
<img width="289" alt="Screenshot 2024-04-16 at 15 40 27" src="https://github.com/zed-industries/zed/assets/121750393/67539645-b019-4f06-b6a7-a3d4160ca12f">

**Comment by Moshyfawn** [comment]
- Related #10454
- Related #8032

**Comment by JosephTLyons** [comment]
Just as an update, we do have a preview button now for markdown files:

![image](https://github.com/user-attachments/assets/253f5162-af3a-4395-8189-34a2cc245ead)

**Comment by iamnbutler** [comment]
There are a number of asks in here, it would help if these were broken down in to a list, and ideally have one ask per issue.

It's hard for this to be super actionable as it is.

As mentioned above, markdown has a preview button already.

> Monitor file changes and serve a directory isn't hard for html files

This feels more in the realm of an extension. Additionally, since Zed is not electron/web based, we would likely have to do some pretty significant work to make this work.

**Comment by FloppyDisco** [comment]
where is the preview button for markdown files??
it does not show up on my editor that i can find

**Comment by frederik-uni** [comment]
the table renderer from the key binds could be used for csv

**Comment by xgfone** [comment]
It works fine with Markdown and SVG files, but not with HTML.

**Reply by xgfone** [reply]
> It works fine with Markdown and SVG files, but not with HTML.

```
Zed Preview 0.218.3 
4adf4c6c1dc3cad2cb35273ee707812c6874ac3b
```

**Comment by malikdraz** [comment]
Concrete HTML preview use case to add to this request:

I am increasingly using agent workflows to generate short-lived, standalone `.html` files for local review: code-review boards, implementation option matrices, design critiques, status reports, and throwaway working documents. These files are intentionally local artifacts. They are usually a single `.html` file with inline CSS and sometimes small inline JS, no deploy step, no web server, and no need for browser devtools.

The workflow I want in Zed:

- Open a local `.html` file and get the same kind of Preview button that Markdown, SVG, and CSV now get.
- Support `Open Preview` and `Open Preview to the Side`, so the source and rendered artifact can be inspected together.
- Render standalone HTML directly from the file or current buffer without requiring a dev server.
- Refresh on save or provide a manual refresh action.
- Keep the source editable and the rendered view read-only.
- Make the security model explicit: local-file preview, no publishing, no hidden upload/share behavior, and either sandbox or clearly gate network/storage/script behavior.

This is narrower than "add a full browser to Zed." The main value is removing the browser context switch for agent-generated local review artifacts and documentation specimens where Zed is already the place the file was created.

**Comment by naanoo** [comment]
+1 Would love HTML preview.

30 % of my output is HTML.

**Comment by kazuki-N5** [comment]
I understand wanting to keep the editor lightweight, but I think an HTML preview is a must-have.
